store/src/main/java/org/apache/rocketmq/store/DefaultMessageStore.java

```diff
@@ -764,7 +764,7 @@ public GetMessageResult getMessage(final String group, final String topic, final
 
                             boolean isInDisk = checkInDiskByCommitOffset(offsetPy, maxOffsetPy);
 
-                            if (cqUnit.getQueueOffset() - offset > maxFilterMessageCount) {
+                            if ((cqUnit.getQueueOffset() - offset) * ConsumeQueue.CQ_STORE_UNIT_SIZE > maxFilterMessageCount) {
```

Comment:
IMO, it's better to modify `maxFilterMessageCount` like this
```java
final int maxFilterMessageCount = Math.max(800 /* 16000 / 20 */, maxMsgNums);
```

There are 2 kinds of ConsumeQueue in RocketMQ 5.0
`ConsumeQueue` which unit size is 20, and `BatchConsumeQueue` which unit size is 46.
`(cqUnit.getQueueOffset() - offset) * ConsumeQueue.CQ_STORE_UNIT_SIZE` may be confused

Reply:
> For some pulled messages for maxMsgNums could be filter-out. so `Message Count ` is not a good way to limit. Change Count-Limit to Bytes-Limit, you can check code on 4.9.x.
> 
> https://github.com/apache/rocketmq/blob/c88686f13e6f4acef1f00c6d3e18589683ec3368/store/src/main/java/org/apache/rocketmq/store/DefaultMessageStore.java#L623
> 
> but it seems need some updates on
> 
> ```java
> if ((cqUnit.getQueueOffset() - offset) *  CQ_STORE_UNIT_SIZE  > maxFilterMessageCount) {
>     break;
> }
> ```
I'm agree with HScarb cause UNIT_SIZE will be different between BatchConsumeQueue and ConsumeQueue. @francisoliverlee so I wander why we have to use Byte-Limit rather than Count-Limit

Reply:
count-based comparison is more understandable. If there are implementation-specific subtle differences, there should be a concept to abstract away from them.